File: Python/dsaOne/Blind 75/Javascript/Arrays_and_Hashing/productOfArrayExceptSelf.js
const productOfArrayExceptSelf = (nums) => {
    let res = []
    let prefix = 1
    let postfix = 1

    for(let i = 0; i < nums.length; i++) {
        res[i] = prefix
        prefix *= nums[i]
    }

    for(let j = res.length - 1; j > -1; j--) {
        res[j] *= postfix
        postfix *= nums[j]
    }

    return res
}

console.log(productOfArrayExceptSelf([1,2,3,4])) //[24,12,8,6]

File: Python/dsaOne/Blind 75/Arrays_and_Hashing/productOfArrayExceptSelf.py
#https://leetcode.com/problems/product-of-array-except-self/

def productOfArrayExceptSelf(nums):
    res = []
    prefix = 1
    postfix = 1

    for i in nums:
        res.append(1)

    for i in range(len(nums)):
        res[i] = prefix
        prefix *= nums[i]
    
    for i in range(len(nums)-1,-1,-1):
        res[i] *= postfix
        postfix *= nums[i]
    
    return res

print(productOfArrayExceptSelf([1,2,3,4]))
print(productOfArrayExceptSelf([-1,1,0,-3,3]))